crates/sargon-uniffi/src/signing/signed_outcome.rs

```diff
@@ -0,0 +1,74 @@
+use crate::prelude::*;
+use paste::paste;
+use sargon::SigningAbandonedReason as InternalSigningAbandonedReason;
+
+#[derive(Clone, PartialEq, Eq, InternalConversion, uniffi::Enum)]
+pub enum SigningAbandonedReason {
+    /// The user rejected the signing process
+    Rejected,
+
+    /// The signing process started with no profile present
+    ProfileMissing,
+
+    /// Preprocessing of signatures collector state failed.
+    PreprocessingError(String),
+
+    /// Could not validate signatures
+    InvalidSignatures,
+}
+
+macro_rules! decl_signed_outcome {
+    (signable: $signable:ty, signed: $signed:ty) => {
+        paste! {
+            use sargon::[< $signable >] as [< Internal $signable >];
+            use sargon::[< $signed >] as [< Internal $signed >];
+
+            type [< InternalSignedOutcomeOf $signable >] = sargon::SignedOutcome<[< Internal $signable >]>;
+
+            /// Outcome of signing a transaction intent
+            #[derive(Clone, PartialEq, Eq, uniffi::Enum)]
+            pub enum [< SignedOutcomeOf $signable >] {
+                /// The user has provided all needed signatures, the transaction intent is considered signed
```

Comment:
Eight occurences of ` [< SignedOutcomeOf $signable >] ` here and 6 `[< InternalSignedOutcomeOf $signable >]` maybe wanna employ:
https://github.com/radixdlt/sargon/pull/288/files#r1869189629

technique?